common/src/main/java/org/mcaccess/minecraftaccess/utils/NarrationUtils.java

```diff
@@ -265,60 +282,59 @@ public static Pair<String, String> narrateBlockForContentChecking(BlockPos pos,
-            }
+        if (blockState.isOf(Blocks.WATER) || blockState.isOf(Blocks.LAVA)) {
+            toSpeak = NarrationUtils.narrateFluidBlock(blockPos);
+            return new Pair<>(toSpeak, toSpeak);
+        }
 
-            if (blockEntity != null) {
-                // The all signs tag include all types of signs, so it should also work with the hanging signs in 1.20.x
-                if (blockState.isIn(BlockTags.ALL_SIGNS)) {
-                    toSpeak = getSignInfo((SignBlockEntity) blockEntity, client.player, toSpeak);
-                } else if (blockEntity instanceof BeehiveBlockEntity beehiveBlockEntity) {
-                    Pair<String, String> beehiveInfo = getBeehiveInfo(beehiveBlockEntity, blockState, toSpeak, currentQuery);
-                    toSpeak = beehiveInfo.getLeft();
-                    currentQuery = beehiveInfo.getRight();
+        if (blockEntity != null) {
+            if (blockState.isIn(BlockTags.ALL_SIGNS)) {
+                toSpeak = getSignInfo((SignBlockEntity) blockEntity, client.player, toSpeak);
+            } else if (blockEntity instanceof BeehiveBlockEntity beehiveBlockEntity) {
+                Pair<String, String> beehiveInfo = getBeehiveInfo(beehiveBlockEntity, blockState, toSpeak, currentQuery);
+                toSpeak = beehiveInfo.getLeft();
+                currentQuery = beehiveInfo.getRight();
+            } else
+// Speak monster spawner mob type
+                if (blockEntity instanceof MobSpawnerBlockEntity spawner) {
+                    // Will not support non-vanilla custom configured multiple-mob spawner (like generated with command)
+                    Entity entity = ((MobSpawnerLogicAccessor) spawner.getLogic()).getRenderedEntity();
+                    // Monster spawners that gotten from creative creating screen is empty.
+                    String entityName = "Empty";
+                    if (entity != null) {
+                        entityName = Objects.requireNonNull(entity.getDisplayName()).getString();
+                    }
+                    toSpeak = entityName + " " + toSpeak;
+                    currentQuery = entityName + currentQuery;
                 }
-            }
+        }
 
-            if (block instanceof PlantBlock || block instanceof CocoaBlock) {
-                Pair<String, String> cropsInfo = getCropsInfo(block, blockState, toSpeak, currentQuery);
-                toSpeak = cropsInfo.getLeft();
-                currentQuery = cropsInfo.getRight();
-            }
+        if (block instanceof PlantBlock || block instanceof CocoaBlock) {
+            Pair<String, String> cropsInfo = getCropsInfo(block, blockState, toSpeak, currentQuery);
+            toSpeak = cropsInfo.getLeft();
+            currentQuery = cropsInfo.getRight();
+        } else
 
             // Check if farmland is wet
             if (block instanceof FarmlandBlock && blockState.get(FarmlandBlock.MOISTURE) == FarmlandBlock.MAX_MOISTURE) {
```

Comment:
Else ifs shouldn't have multiple linebreaks between the `else` and `if`
```suggestion
        } else if (block instanceof FarmlandBlock && blockState.get(FarmlandBlock.MOISTURE) == FarmlandBlock.MAX_MOISTURE) {
```